include/irr/asset/CTexturePacker.h

```diff
@@ -145,16 +119,12 @@ class ITexturePacker
             (*offsets)[i] = (addr==CSquareAddressAllocator<uint32_t>::invalid_address) ? 
                 STexOffset::invalid_addr :
                 STexOffset(m_addrAlctr.unpackAddress_x(addr), m_addrAlctr.unpackAddress_y(addr), m_addrAlctr.unpackAddress_layer(addr));
-
-            //i didnt know where to put page table page table and filling code so it is here for now
-            uint32_t* const pgTab = reinterpret_cast<uint32_t*>(
-                reinterpret_cast<uint8_t*>(m_pageTable->getBuffer()->getPointer()) + m_pageTable->getRegions().begin()[i].bufferOffset
-                );
-            const uint32_t pitch = m_pageTable->getRegions().begin()[i].bufferRowLength;
-            const uint32_t offset = (pitch*(m_tilesPerDim>>i))*(*offsets)[i].layer() + (*offsets)[i].y()*pitch + (*offsets)[i].x();
-            for (uint32_t x = 0u; x < pgSqrSz; ++x)
-                for (uint32_t y = 0u; y < pgSqrSz; ++y)
-                    pgTab[offset + y*pitch + x] = (*offsets)[i].addr;
+        }
+        if (levelCount > _subres.levelCount)
```

Comment:
this will always happen, the texture will always have a mip-tail < pageSize

unless its clipped, easy to figure out

`any((_img->getMipSize(_subres.baseMipLevel+_subres.levelCount-1u) < m_pgSzxy_as_vectorSIMDu) & mask1100)`

Reply:
> unless its clipped

thats why theres this `if ` :P

Reply:
I don't think that `levelCount` if and logic behind calculating it works out the same.

Reply:
?

Reply:
i dont understand

Reply:
its convoluted and not pretty to work out the `levelCount`, unlike
`any((_img->getMipSize(_subres.baseMipLevel+_subres.levelCount-1u) < m_pgSzxy_as_vectorSIMDu) & mask1100)`

Reply:
Moved to:
https://github.com/buildaworldnet/IrrlichtBAW/pull/424/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&hide-deleted-files=true#r399560955